fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/cursors/CursorLimitManager.java

```diff
@@ -136,7 +136,7 @@ public boolean tryRecordScan() {
         if (!halted) {
             usedInitialPass = true;
         } else if (failOnScanLimitReached) {
-            throw new ScanLimitReachedException("limit on number of key-values scanned per transaction reached");
+            throw new ScanLimitReachedException("Reached cursor limit : " + (getStoppedReason().isPresent() ? getStoppedReason().get().toString() : "Unknown reason."));
```

Comment:
With very rare exceptions, all `RecordCoreExceptions` are expected to have static title strings.  Any information that is specific to the action being performed should be included as logging details.  So you want to do something like:

    throw new ScanLimitReachedException("limit on number of key-values scanned per transaction reached")
        .addLogInfo("no_next_reason", getStoppedReason().orElse("Unknown"));
    
We do this because it allows for easier log analysis.